chain/jsonrpc-primitives/src/types/query.rs

```diff
@@ -0,0 +1,204 @@
+use serde_json::Value;
+
+/// Max size of the query path (soft-deprecated)
+const QUERY_DATA_MAX_SIZE: usize = 10 * 1024;
+
+#[derive(Serialize, Deserialize)]
+pub struct RpcQueryRequest {
+    #[serde(flatten)]
+    pub block_reference: near_primitives::types::BlockReference,
+    #[serde(flatten)]
+    pub request: near_primitives::views::QueryRequest,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RpcQueryError {
+    #[error("IO Error: {error_message}")]
+    IOError { error_message: String },
+    #[error("There are no fully synchronized blocks on the node yet")]
+    NoSyncedBlocks,
+    #[error("The node does not track the shard")]
+    UnavailableShard { requested_shard_id: near_primitives::types::ShardId },
+    #[error("Account ID {requested_account_id} is invalid")]
+    InvalidAccount {
+        requested_account_id: near_primitives::types::AccountId,
+        block_height: near_primitives::types::BlockHeight,
+        block_hash: near_primitives::hash::CryptoHash,
+    },
+    #[error("account {requested_account_id} does not exist while viewing")]
+    UnknownAccount {
+        requested_account_id: near_primitives::types::AccountId,
+        block_height: near_primitives::types::BlockHeight,
+        block_hash: near_primitives::hash::CryptoHash,
+    },
+    #[error(
+        "Contract code for contract ID #{contract_account_id} has never been observed on the node"
+    )]
+    NoContractCode {
+        contract_account_id: near_primitives::types::AccountId,
+        block_height: near_primitives::types::BlockHeight,
+        block_hash: near_primitives::hash::CryptoHash,
+    },
+    #[error("Access key for public key {public_key} has never been observed on the node")]
+    UnknownAccessKey {
+        public_key: near_crypto::PublicKey,
+        block_height: near_primitives::types::BlockHeight,
+        block_hash: near_primitives::hash::CryptoHash,
+    },
+    #[error("Function call returned an error: {vm_error}")]
+    ContractExecutionError {
+        vm_error: String,
+        block_height: near_primitives::types::BlockHeight,
+        block_hash: near_primitives::hash::CryptoHash,
+    },
+    #[error("The node reached its limits. Try again later. More details: {error_message}")]
+    InternalError { error_message: String },
+    // NOTE: Currently, the underlying errors are too broad, and while we tried to handle
+    // expected cases, we cannot statically guarantee that no other errors will be returned
+    // in the future.
+    // TODO #3851: Remove this variant once we can exhaustively match all the underlying errors
+    #[error("It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose. Details: {error_message}")]
+    Unreachable { error_message: String },
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RpcQueryResponse {
+    #[serde(flatten)]
+    pub query_response: near_primitives::views::QueryResponse,
+}
```

Comment:
I feel that the nesting is completely unnecessary here. We should either (1) implement our own `RpcQueryResponse` or (2) just type-alias the `views::QueryResponse`. For clear separation of concerns and following our previous style, we should consider (1) and also removing the serde serialization from `view::QueryResponse`.